plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -71,17 +73,26 @@ public HudiSplitSource(
             int maxOutstandingSplits,
             List<String> partitions)
     {
-        HudiTableMetaClient metaClient = buildTableMetaClient(fileSystemFactory.create(session), tableHandle.getBasePath());
+        boolean enableMetadataTable = isHudiMetadataTableEnabled(session);
+        HoodieTableMetaClient metaClient = buildTableMetaClient(fileSystemFactory.create(session), tableHandle.getBasePath());
+        String latestCommitTime = metaClient.getActiveTimeline()
+                .getCommitsTimeline()
+                .filterCompletedInstants()
+                .lastInstant()
+                .map(HoodieInstant::getTimestamp)
+                .orElseThrow(() -> new TrinoException(HudiErrorCode.HUDI_NO_VALID_COMMIT, "Table has no valid commits"));
         List<HiveColumnHandle> partitionColumnHandles = table.getPartitionColumns().stream()
                 .map(column -> partitionColumnHandleMap.get(column.getName())).collect(toList());
 
-        HudiDirectoryLister hudiDirectoryLister = new HudiReadOptimizedDirectoryLister(
+        HudiDirectoryLister hudiDirectoryLister = new HudiSnapshotDirectoryLister(
```

Comment:
Is `HudiReadOptimizedDirectoryLister` not required anymore? I think we can have a session prop for query_type and then decide which directory lister to use

Reply:
I think we can have a single `HudiDirectoryLister` and determine the splits to return within the same class so it's easier to maintain.  I'll refactor this part.

Reply:
makes sense